server/src/test/java/org/apache/iotdb/db/metadata/MManagerBasicTest.java

```diff
@@ -957,6 +962,46 @@ public void testTemplateInnerTree() {
     }
   }
 
+  @Test
+  public void testCreateSchemaTemplateSerializationAdaptation() throws IOException {
+    CreateTemplatePlan plan = getCreateTemplatePlan();
+    ByteArrayOutputStream baos = new ByteArrayOutputStream();
+    DataOutputStream dos = new DataOutputStream(baos);
+    plan.formerSerialize(dos);
+    byte[] byteArray = baos.toByteArray();
+    ByteBuffer buffer = ByteBuffer.wrap(byteArray);
+
+    assertEquals(PhysicalPlan.PhysicalPlanType.CREATE_TEMPLATE.ordinal(), buffer.get());
+
+    CreateTemplatePlan deserializedPlan = new CreateTemplatePlan();
+    deserializedPlan.deserialize(buffer);
+
+    assertEquals(plan.getCompressors().size(), deserializedPlan.getCompressors().size());
+    assertEquals(
+        plan.getMeasurements().get(0).get(0), deserializedPlan.getMeasurements().get(0).get(0));
+    assertEquals(plan.getDataTypes().size(), deserializedPlan.getDataTypes().size());
+  }
+
+  @Test
+  public void testCreateSchemaTemplateSerialization() throws IOException {
+    CreateTemplatePlan plan = getCreateTemplatePlan();
+    ByteArrayOutputStream baos = new ByteArrayOutputStream();
+    DataOutputStream dos = new DataOutputStream(baos);
+    plan.serialize(dos);
+    byte[] byteArray = baos.toByteArray();
+    ByteBuffer buffer = ByteBuffer.wrap(byteArray);
+
+    assertEquals(PhysicalPlan.PhysicalPlanType.CREATE_TEMPLATE.ordinal(), buffer.get());
+
+    CreateTemplatePlan deserializedPlan = new CreateTemplatePlan();
+    deserializedPlan.deserialize(buffer);
+
+    assertEquals(
+        plan.getCompressors().get(0).get(0), deserializedPlan.getCompressors().get(0).get(0));
+    assertEquals(plan.getMeasurements().size(), deserializedPlan.getMeasurements().size());
+    assertEquals(plan.getName(), deserializedPlan.getName());
+  }
```

Comment:
This test only concerns createTemplatePlan and doesn't involves any MManager action. I think it's better to remove it from MManagerBasicTest

Reply:
these tests now moved to TemplateTest.java under metadata package now, thanks !